docs/guide/data-security.asciidoc

```diff
@@ -0,0 +1,408 @@
+
+* Go: {apm-go-ref-v}/configuration.html#config-capture-headers[`ELASTIC_APM_CAPTURE_HEADERS`]
+* Java: {apm-java-ref-v}/config-core.html#config-sanitize-field-names[`capture_headers`]
+* .NET: {apm-dotnet-ref-v}/config-http.html#config-capture-headers[`CaptureHeaders`]
+* Node.js: {apm-node-ref-v}/configuration.html#capture-headers[`captureHeaders`]
+// * PHP: {apm-php-ref-v}[``]
+* Python: {apm-py-ref-v}/configuration.html#config-capture-headers[`capture_headers`]
+* Ruby: {apm-ruby-ref-v}/configuration.html#config-capture-headers[`capture_headers`]
+
+[discrete]
+[[filter-http-body]]
+==== HTTP bodies
+
+By default, the body of HTTP requests is not recorded.
+Request bodies often contain sensitive data like passwords or credit card numbers,
+so use care when enabling this feature.
+
+This setting supports {kibana-ref}/agent-configuration.html[Central configuration],
+which means the list of sanitized fields can be updated without needing to redeploy your services:
+
+* Go: {apm-go-ref-v}/configuration.html#config-capture-body[`ELASTIC_APM_CAPTURE_BODY`]
+* Java: {apm-java-ref-v}/config-core.html#config-sanitize-field-names[`capture_body`]
+* .NET: {apm-dotnet-ref-v}/config-http.html#config-capture-body[`CaptureBody`]
+* Node.js: {apm-node-ref-v}//configuration.html#capture-body[`captureBody`]
+// * PHP: {apm-php-ref-v}[``]
+* Python: {apm-py-ref-v}/configuration.html#config-capture-body[`capture_body`]
+* Ruby: {apm-ruby-ref-v}/configuration.html#config-capture-body[`capture_body`]
+
+[discrete]
+[[filter-real-user-data]]
+==== Real user monitoring data
+
+Protecting user data is important.
+For that reason, individual RUM instrumentations can be disabled in the RUM agent with the
+{apm-rum-ref-v}/configuration.html#disable-instrumentations[`disableInstrumentations`] configuration variable.
+Disabled instrumentations produce no spans or transactions.
+
+[options="header"]
+|====
+|Disable |Configuration value
+|HTTP requests |`fetch` and `xmlhttprequest
+|Page load metrics including static resources |`page-load`
+|User browser errors |`error`
+|User click events including URLs visited, mouse clicks, and navigation events |`eventtarget`
+|====
+
+[discrete]
+[[filter-database-statements]]
+==== Database statements
+
+For SQL databases, APM agents do not capture the parameters of prepared statements.
+To filter or obfuscate data in non-SQL database statements,
+you can set up an ingest node pipeline.
+
+[discrete]
+[[filter-agent-specific]]
+==== Agent-specific options
+
+Certain agents offer additional filtering and obfuscating options:
+
+**Agent configuration options**
+
+* (Node.js) Remove errors raised by the server-side process:
+Disable with {apm-node-ref-v}/configuration.html#capture-exceptions[captureExceptions].
+
+* (Java) Remove process arguments from transactions:
+* Disabled by default with {apm-java-ref-v}/config-reporter.html#config-include-process-args[`include_process_args`].
```

Comment:
Should we include these? I'll need to dig through the agent configs to find others. Any recommendations?

Reply:
Good call on including these, as process arguments may contain sensitive info. Currently, only the Java agent has that setting. Other agents capture the process args by default. I think we should call this out so that folks are aware of that and can drop the field with an ingest node processor.

I was wondering whether we should maybe even maintain a list/table of potentially sensitive properties to make it easier for folks to review the collected data. Also relates to https://github.com/elastic/apm-server/issues/5087.

Reply:
Thanks, Felix. I added both of your suggestions to #5087. I'll need some help coming up with the list of potentially sensitive properties.